src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public void register(String email, String password) throws RegistrationException {
+        try {
+            User user = new User();
+            user.setEmail(email);
+            user.setPassword(password);
+            userService.add(user);
+        } catch (Exception e) {
+            throw new RegistrationException(
+                    "Can't register new user. Possible cause: user with email " + email
+                            + " already exists");
+        }
```

Comment:
remove this try-catch
retrieve user with userService
compare passwords, throw if they don't match

Reply:
I'm not sure if I understood you correctly, but I just returned my previous solution for the method. Hope it works for you.